<a href="https://colab.research.google.com/github/mikeytw11/Hockey/blob/main/Devils_Start.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#Import dependencies
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [ ]:
#Import 2021-2022 Devils Data for the first 10 games.
Team_Stats_21_22 = pd.read_csv("https://raw.githubusercontent.com/mikeytw11/Hockey/main/Devils%20Start/2021-2022.csv")
#Import 2022-2023 Devils Data for the first 10 games.
Team_Stats_22_23 = pd.read_csv("https://raw.githubusercontent.com/mikeytw11/Hockey/main/Devils%20Start/2022-2023.csv")

Team_Stats_21_22.head(5)

Name Team                Dates  GP     TOI    GF%    SF%    FF%  \
0       Ducks  ANA  10/15/21 - 11/06/21  11  521.10  47.74  47.62  48.46   
1     Coyotes  ARI  10/15/21 - 11/06/21  11  541.07  37.61  42.90  43.27   
2      Bruins  BOS  10/15/21 - 11/06/21   9  436.57  45.52  56.94  56.59   
3      Sabres  BUF  10/15/21 - 11/06/21  10  494.15  44.23  48.96  48.89   
4  Hurricanes  CAR  10/15/21 - 11/06/21   9  389.02  72.50  54.11  54.62   

     CF%   xGF%  ...  CA/60  xGF/60  xGA/60  G±/60  S±/60  F±/60  C±/60  \
0  48.71  47.90  ...  57.83    2.54    2.77  -0.22  -2.89  -2.57  -2.90   
1  42.17  43.23  ...  59.14    1.74    2.28  -1.08  -7.85 -10.35 -16.02   
2  55.57  58.53  ...  47.43    2.41    1.71  -0.35   8.62  10.81  11.89   
3  49.42  47.02  ...  51.39    2.17    2.45  -0.50  -1.25  -1.76  -1.18   
4  54.84  52.97  ...  52.06    2.79    2.47   2.02   5.27   8.06  11.16   

   xG±/60   Sh%    Sv%  
0   -0.22  7.91  92.13  
1   -0.54  6.90  91.40  
2    0.70  5.07  91.98  
3   -0.27  6.51  92.13  
4    0.31  9.36  95.81  

[5 rows x 27 columns]

In [ ]:
#Restrict data for the 2021-2022 season to just the Devils
Devils_21_22 = Team_Stats_21_22.loc[Team_Stats_21_22.Name=="Devils"]
Devils_21_22

#Restrict the date for the 2022-2023 season to just the Devils.


Name Team                Dates  GP    TOI    GF%    SF%    FF%    CF%  \
16  Devils  N.J  10/15/21 - 11/06/21  10  462.4  50.48  53.28  51.93  50.64   

     xGF%  ...  CA/60  xGF/60  xGA/60  G±/60  S±/60  F±/60  C±/60  xG±/60  \
16  52.97  ...  56.45    2.72    2.41   0.04   3.93   3.24   1.47     0.3   

     Sh%    Sv%  
16  6.35  92.89  

[1 rows x 27 columns]

In [ ]:
#Restrict data to GF%, GF/60, GA/60, xGF%, xGF/60, xGA/60, Sh%, Sv%
Devils_21_22 = Devils_21_22.loc[:,['GF%','GF/60','GA/60','xGF%','xGF/60','xGA/60','Sh%','Sv%']]
Devils_21_22

GF%  GF/60  GA/60   xGF%  xGF/60  xGA/60   Sh%    Sv%
16  50.48   2.03   1.99  52.97    2.72    2.41  6.35  92.89

| Metric | 2021-2022 | 2022-2023 |
|:--------------|:---------|:---------|
|Record     | 7-3-2 | 9-3-0 |
|Goal Differential|  +4 | +13 |
| GF % | 54.64% (10th) | 59.34% (5th) |
| GF/60 | 2.41 (14th) | 3.39 (1st)  |
| GA/60 | 2.00 (24th) |  2.32 (10th) |
||
| xGF % | 50.99% (13th) | 64.91% (1st)  |
| xGF/60 | 2.72 (5th) | 3.61 (1st) |
| xGA/60 | 2.62 (25th) | 1.95 (1st) |
||
| Sv % | 93.38% (9th) | 89.38 (32nd)  |
| Sh % | 7.56% (18th) | 8.16 (22nd)  |
||
| PP GF/60 |  4.18 (26th)  |  7.41 (18th)  |
| PP xGF/60 | 4.8 (30th)  |  10.65 (3rd)  | 
||
| PK GA/60 |  8.67 (8th) |  3.77 (4th)  | 
| PK xGA/60 | 4.88 (29th)  |  5.81 (4th)  |